### PR TITLE
FEATURE: ONE-4121 Allow KPIs and Headlines with drills to not to be underlined

### PR DESCRIPTION
**Pull request by ivanmjartan**
…nderlined

---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5039/ : SUCCESS in 5m 44s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/403/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5043/ : SUCCESS in 13m 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/410/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5059/ : SUCCESS in 5m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/433/ : SUCCESS in 10s

**Comment by ivanmjartan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1218.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/436/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2062/ : SUCCESS in 18m 29s

**Comment by ivanmjartan** [comment]
extended test - storybook

File: src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx [hunk 1]
```diff
@@ -221,4 +225,15 @@ export class PluggableHeadline extends AbstractPluggableVisualization {
             return resultItems;
         }, []);
     }
+
+    private setConfigFromFeatureFlags(config: IChartConfig): IChartConfig {
```
Comment (xMort) [inline src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx]: I just checked the PR out of curiosity and this block of code is repeated twice.  
Also, the second `if` is unnecessary.

`if (this.featureFlags && this.featureFlags.disableKpiDashboardHeadlineUnderline === true)`

Maybe I would one lined it as: 

````
return this.featureFlags && this.featureFlags.disableKpiDashboardHeadlineUnderline === true
    ? { ...config, disableDrillUnderline: true }
    : config;
````

(I don't think that functions that modify its parameters are good practice)

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2650/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/438/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2063/ : UNSTABLE in 15m 10s

File: src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx (hunk 1 above)
Comment (ivanmjartan) [reply src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx]: Thx for comment. Fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5064/ : SUCCESS in 7m 46s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/440/ : SUCCESS in 10s

**Comment by ivanmjartan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5065/ : SUCCESS in 5m 11s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/441/ : SUCCESS in 6s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2651/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/442/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2064/ : SUCCESS in 14m 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5067/ : SUCCESS in 4m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/444/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- gooddata-react-components-unit-tests-zuul-docker-el7 : SKIPPED
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/447/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 3s

**Comment by ivanmjartan** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- gooddata-react-components-unit-tests-zuul-docker-el7 : SKIPPED
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/448/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 1s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- gooddata-react-components-unit-tests-zuul-docker-el7 : SKIPPED
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/449/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 2s

File: src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx (hunk 1 above)
Comment (xMort) [reply src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx]: You still have it here twice ;) imho
src/components/uri/Visualization.tsx:369

Comment (ivanmjartan) [reply src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx]: Ok I will move this method to service

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5076/ : FAILURE in 1m 40s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/456/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5077/ : SUCCESS in 5m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/457/ : SUCCESS in 7s

**Comment by xMort** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2654/ : SUCCESS in 11m 48s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5082/ : SUCCESS in 4m 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1252/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1667/ : SUCCESS in 15m 21s

**Pull request merged**